### ISSUE TITLE
Track model configuration in addition to hyperparamters

### ISSUE THREAD

**Issue by Uroc327** [PyTorchLightning/pytorch-lightning#1736]
## 🚀 Feature
I would like to separate the current `hparams` (for example used to log models) into two groups:
- Hyperparamters: are those hparams, which can be optimized without changing the models meaning and may change the performance. For example the amount of samples generated, the batch size or the learning rate.
- Configuration: are those hparams, which actually change the 'meaning' of the model and can not be optimized in a useful, automatic manner. Possible examples are the noise variance for a denoising task or a random seed for reproducible noise generation.

### Motivation

I'd like to see both types of hparams in my logging, in order to be able to tell which model ran with what set of configuration. I think, separating actual hyperparamters from configuration also makes it easier to optimizer hyperparameters automatically.

### Pitch

I guess, the easiest way would be to duplicate (maybe slightly alter) the current behavior of the `LightningModule.hparams` field in an additional field (e.g. `LightningModule.config`).
